src/main/iidxhook9/fs-hook.c

```diff
@@ -0,0 +1,50 @@
+#define LOG_MODULE "fs-hook"
+
+#include <stdint.h>
+#include <string.h>
+
+#include "hook/table.h"
+
+#include "imports/avs.h"
+
+#include "iidxhook9/fs-hook.h"
+
+#include "util/log.h"
+#include "util/str.h"
+
+static void *(*real_avs_fs_mount)(const char *dest, const char *src, const char *fs_type, const char *options);
+static void *my_avs_fs_mount(const char *dest, const char *src, const char *fs_type, const char *options);
+
+static const struct hook_symbol avs_fs_hook_syms[] = {
+    {.name = "XCgsqzn000004b", // avs_fs_mount
```

Comment:
suggestion: I guess it's fine to hook it like that, but that might introduce re-usability issues with different AVS versions. I suggest to at least ifdef guard it and check that the AVS_VERSION define is within the right version range. Probably just throw an error for now if it isn't, so the next dev has visibility on that matter at least and needs to double check.

Reply:
The hook is scoped to iidxhook9, so the symbol will always be that.

Reply:
Right, thanks for highlighting. Kinda missed that detail.